source/common/buffer/buffer_impl.h

```diff
@@ -764,24 +756,20 @@ class OwnedImpl : public LibEventInstance {
   uint8_t* inlineReserve(uint64_t size) override {
     if (slices_.empty()) {
       slices_.emplace_back(Slice(size, account_));
-      return slices_.back().inlineReserve(size);
+      length_ += size;
```

Comment:
can you add some comments describing the two cases here?

Reply:
git it.